doc/neps/nep-0041-improved-dtype-support.rst

```diff
@@ -0,0 +1,715 @@
+datatypes:
+
+* Creating casting rules for parametric user-defined dtypes is either impossible
+  or so complex that it has never been attempted.
+* Type promotion, e.g. the operation deciding that adding float and integer
+  values should return a float value, is very valuable for numeric datatypes
+  but is limited in scope for user-defined and especially parametric datatypes.
+* Much of the logic (e.g. promotion) is written in single functions
+  instead of being split as methods on the datatype itself.
+* In the current design datatypes cannot have methods that do not generalize
+  to other datatypes. For example a unit datatype cannot have a ``.to_si()`` method to
+  easily find the datatype which would represent the same values in SI units.
+
+The large need to solve these issues has driven the scientific community
+to create work-arounds in multiple projects implementing physical units as an
+array-like class instead of a datatype, which would generalize better across
+multiple array-likes (Dask, pandas, etc.).
+Already, Pandas has made a push into the same direction with its
+extension arrays [pandas_extension_arrays]_ and undoubtedly
+the community would be best served if such new features could be common
+between NumPy, Pandas, and other projects.
+
+Scope
+^^^^^
+
+The proposed refactoring of the datatype system is a large undertaking and
+thus is proposed to be split into various phases, roughly:
+
+* Phase I: Restructure and extend the datatype infrastructure (This NEP 41)
+* Phase II: Incrementally define or rework API (Detailed largely in NEPs 42/43)
+* Phase III: Growth of NumPy and Scientific Python Ecosystem capabilities.
+
+For a more detailed accounting of the various phases, see
+"Plan to Approach the Full Refactor" in the Implementation section below.
+This NEP proposes to move ahead with the necessary creation of new dtype
+subclasses (Phase I),
+and start working on implementing current functionality.
+Within the context of this NEP all development will be fully private API or
+use preliminary underscored names which must be changed in the future.
+Most of the internal and public API choices are part of a second Phase
+and will be discussed in more detail in the following NEPs 42 and 43.
+The initial implementation of this NEP will have little or no effect on users,
+but provides the necessary ground work for incrementally addressing the
+full rework.
+
+The implementation of this NEP and the following, implied large rework of how
+datatypes are defined in NumPy is expected to create small incompatibilities
+(see backward compatibility section).
+However, a transition requiring large code adaption is not anticipated and not
+within scope.
+
+Specifically, this NEP makes the following design choices which are discussed
+in more details in the detailed description section:
+
+1. Each datatype will be a class instance of a DType class, with most of the
```

Comment:
Are you trying to say:
```suggestion
1. Each datatype will be an instance of a subclass of ``np.dtype``, with most of the
```
or is this phrasing deliberately different.

Reply:
No, that is much better to understand, thanks.